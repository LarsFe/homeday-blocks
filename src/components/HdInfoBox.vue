<template>
  <section class="info-box">
    <img
      :src="icon"
      class="info-box__icon"
    >
    <p
      class="info-box__message"
      v-html="message"
    />
  </section>
</template>

<script>
export default {
  name: 'HdInfoBox',
  props: {
    message: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      // eslint-disable-next-line global-require
      icon: require('hd-blocks/assets/icons/ic_info.svg'),
    };
  },
};
</script>

<style lang="scss">
@import 'hd-blocks/styles/mixins.scss';

.info-box {
  display: flex;
  align-items: flex-start;
  padding-bottom: $stack-s;
  border-bottom: 1px solid $heather;
  color: $nevada;

  &__icon {
    margin: $stack-xs $inline-s $stack-xs $stack-xs;
  }

  &__message {
    @include font('text-xsmall');

    margin-top: $stack-xs;

    @media (min-width: $break-tablet) {
      @include font('text-small');

      margin-top: 0;
    }

    a {
      color: inherit;
    }
  }
}
</style>
