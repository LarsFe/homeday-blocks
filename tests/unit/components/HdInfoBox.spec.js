import { mount } from '@vue/test-utils';
import HdInfoBox from '@/components/HdInfoBox.vue';

const MESSAGE = 'Hello, Homeday!';
const MESSAGE_SELECTOR = '.info-box__message';
describe('HdInfoBox', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(HdInfoBox, {
      propsData: {
        message: MESSAGE,
      },
    });
  });

  test('renders component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders a message correctly', () => {
    expect(wrapper.find(MESSAGE_SELECTOR).text()).toBe(MESSAGE);
  });
});
