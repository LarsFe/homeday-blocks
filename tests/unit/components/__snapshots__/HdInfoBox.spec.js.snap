// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdInfoBox renders component 1`] = `
<section class="info-box"><img src="" class="info-box__icon">
  <p class="info-box__message">Hello, Homeday!</p>
</section>
`;
